src/array_api_extra/_lib/_compat.pyi

```diff
@@ -0,0 +1,13 @@
+from types import ModuleType
+
+from ._typing import Array, Device
+
+class ArrayModule(ModuleType):
+    def device(self, x: Array, /) -> Device: ...
+
+def array_namespace(
+    *xs: Array,
+    api_version: str | None = None,
+    use_compat: bool | None = None,
+) -> ArrayModule: ...
+def device(x: Array, /) -> Device: ...
```

Comment:
This is a temporary hack to make mypy AND pyright happy. In the long run we should make array_api_compat typed.